react-native.config.js

```diff
@@ -0,0 +1,68 @@
+//
+// Copyright (c) Microsoft Corporation
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+//
+// @ts-check
+
+/**
+ * @template ConfigT
+ * @typedef {{
+ *   name: string,
+ *   description?: string,
+ *   func: (argv: Array<string>, config: ConfigT, args: {}) => ?Promise<void>,
+ *   options?: Array<{
+ *     name: string,
+ *     description?: string,
+ *     parse?: (val: string) => any,
+ *     default?:
+ *       | string
+ *       | boolean
+ *       | number
+ *       | ((config: ConfigT) => string | boolean | number),
+ *   }>,
+ *   examples?: Array<{
+ *     desc: string,
+ *     cmd: string,
+ *   }>,
+ * }} Command;
+ */
+
+/** @type {{ commands: Command<{}>[] }} */
+module.exports = {
+  commands: [
+    {
+      name: "init-test-app",
+      description: "Initializes a new test app project",
+      func: (_, __, args) => {
+        const path = require("path");
+        const plop = require("node-plop");
+        // @ts-ignore tsc doesn't think that "node-plop" returns a function
+        plop(path.join(__dirname, "plopfile.js"), {
+          destBasePath: args["destination"],
+        })
+          .getGenerator("app")
+          .runActions(args);
+      },
+      options: [
+        {
+          name: "--destination [string]",
+          description: "Path to the destination directory",
+          default: process.cwd(),
+        },
+        {
+          name: "--name [string]",
+          description: "Name of the test app",
+          default: "ReactTestApp",
+        },
+        {
+          name: "--platforms [string]",
+          description:
+            "Platforms to add support for: all, android, ios, macos, windows",
```

Comment:
Is this a comma-separated list? If so, it would be good to state that. E.g.

```suggestion
            "Comma-separated list of platforms to support: all, android, ios, macos, windows",
```

Reply:
This is supposed to be "pick one" option. Any suggestions on how to phrase that?

Reply:
I see. I’d rename the option to be singular and perhaps say something like:

```suggestion
            "Specific platform to support: all, android, ios, macos, windows",
```

Having said that, if you _can_ target `all`, then shouldn’t you be able to pick and permutation too?

(Also, is there a way to infer the permutation from the configuration of the repo?)

Reply:
> Having said that, if you _can_ target `all`, then shouldn’t you be able to pick any permutation too?
> (Also, is there a way to infer the permutation from the configuration of the repo?)

Created issue #233.

I think it's a good idea. We could probably use `react-native config` to detect supported platforms. But I'm hesitant to make such a change in this PR. For now, I think "all" is a reasonable default.